main.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/grandcat/zeroconf"
+)
+
+const (
+	cptvFolder       = "/var/spool/cptv/downloaded"
+	avahiServiceType = "_cacophonator-management._tcp"
+	ledTriggerFile   = "/sys/class/leds/led0/trigger"
+)
+
+type device struct {
+	Name string
+	IPv4 string
+	Port int
+}
+
+func (d device) getRecordingsList() []string {
+	resp, err := http.Get(d.getAddr() + "/api/recordings")
+	if err != nil {
+		log.Println(err)
+		return nil
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		log.Println("non 200 response when getting recordings list")
+		return nil
+	}
+	var ids []string
+	decoder := json.NewDecoder(resp.Body)
+	if err := decoder.Decode(&ids); err != nil {
+		log.Println(err)
+		return nil
+	}
+	return ids
+}
```

Comment:
It's generally a good idea to return errors instead of just logging. This makes the software more flexible (and easier to test). The errors here should be returned instead of logged.